### PR TITLE
STAC harmonization changes

### LINKED ISSUE #178: STAC / Records Harmonization
There was a discussion at the Sept 2022 OGC Metadata Code Sprint about how Records and STAC could come into closer alignment and the conclusion of that discuss was:

- Remove the `created` and `updated` fields that describe when a resource was created and updated.
- Rename the `recordCreated` and `recordUpdated` fields to simply be `created` and `updated`
- Modify the `link` structure to add optional `created` and `updated` fields

To understand the implication, imgine that you have a dataset.  That dataset has several _distributions_ (for the sake of example say a ZIP and TGZ distribution).  A record is created to describe the dataset (and the `created` and `updated` properties for the record are populated accordingly).  Links are added to the `links` section of the record that point to the two distribution and the `created` and `updated` fields in each link is populated to reflect when each distribution was created and last updated.

NOTE: This issue was created to describe the conclusions for the code sprint discussion and as an anchor for a PR that I am working on to implement the agreed upon changes.

### LINKED ISSUE #231: Add more guidance about the `license` field.
In the description of the `license` member make sure to mention:

-  that implementers should NOT include the full text of the license as the value of this field.  This field is meant to be a "code" that stands for the license (e.g. an SPDX code).
- normally, if you include a code to a license then there should also be a link (`rel=license`) that points to the license.  In some cases however (e.g. the `license` member value is an SPDX code) this is not necessary since such licenses are well known.  So, if the `license` member value is an SPDX code should we require a link or not?  STAC does not require a link is the `license` field is an SPDX code BUT it is also not an issue if it is there.
- personally, I am happy to do what STAC does in this instance.

### LINKED ISSUE #232: Fix time example
Following up from #189 I see that we have a definition of time parameter [here](https://github.com/opengeospatial/ogcapi-records/blob/master/core/openapi/schemas/recordGeoJSON.yaml#L21-L36) but the records [example](https://github.com/opengeospatial/ogcapi-records/blob/master/core/examples/json/record.json) does not include the required time property.

Was time supposed to be similar to [temporal extent](https://github.com/opengeospatial/ogcapi-records/blob/master/core/examples/json/record.json#L126-L135)?

### PR DESCRIPTION
**Pull request by pvretano**
Numerous changes listed in issue #178 to bring OARec and STAC into closer alignment.

Closes #178
Closes #231 
Closes #232

### CODE REVIEW

**Comment by pvretano** [comment]
Closes #232

File: core/openapi/schemas/time.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+oneOf:
+- type: 'null'
+- type: object
+  properties:
```
Comment (tomkralidis) [inline core/openapi/schemas/time.yaml:4]: Can we add `resolution` as an optional property?

```yaml
      resolution:
        description: |-
          Minimum time period resolvable in the dataset, as an ISO 8601 duration
          type: string
```

See #147/#149

**Review by tomkralidis — CHANGES_REQUESTED** [review]

File: core/openapi/schemas/time.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+oneOf:
+- type: 'null'
```
[the remark this answers is not in the document]
Comment (pvretano) [reply core/openapi/schemas/time.yaml]: Fixed.

**Review by kalxas — APPROVED** [review]

**Comment by cnreediii** [comment]
@tomkralidis what does resolution mean in this context? Thnx

**Pull request merged**